### PR TITLE
Fixed bugs in failure cases when saving preferences in Rx and SM

### PR DESCRIPTION
**Pull request by U-DON**
### Changes
* Fixed stuck loading indicator when saving Rx preferences and getting error response.
    * Resolves department-of-veterans-affairs/vets.gov-team#2766.
* Show specific error when saving preferences with blank email. Resolves department-of-veterans-affairs/vets.gov-team#2790.
    > screen shot 2017-05-15 at 8 24 54 pm
* Show specific error when saving preferences with invalid email. Resolves department-of-veterans-affairs/vets.gov-team#2791.
    > screen shot 2017-05-15 at 8 25 08 pm

### CODE REVIEW

**Comment by U-DON** [comment]
CC: @NatalieEMoore for awareness on additional error messaging.

File: src/js/messaging/actions/preferences.js [hunk 1]
```diff
@@ -21,9 +21,9 @@ export function fetchPreferences() {
     apiRequest(
       baseUrl,
       null,
-      data => dispatch({
+      response => dispatch({
         type: SM_FETCH_PREFERENCES_SUCCESS,
-        preferences: data.data.attributes
+        preferences: response.data.attributes
```
Comment (bshyong) [inline src/js/messaging/actions/preferences.js]: 👍  on clarity

File: src/js/messaging/utils/helpers.js [hunk 2]
```diff
@@ -47,15 +47,17 @@ export function apiRequest(resource, optionalSettings = {}, success, error) {
...
+                 ? response.json()
+                 : Promise.resolve(response);
+
       if (!response.ok) {
         // Refresh to show login view when requests are unauthorized.
         if (response.status === 401) { return window.location.reload(); }
-        return Promise.reject(response);
-      } else if (isJson(response)) {
-        return response.json();
+        return data.then(Promise.reject.bind(Promise));
```
Comment (bshyong) [inline src/js/messaging/utils/helpers.js]: What is this `bind` for?

**Review by bshyong — APPROVED** [review]
LGTM!

**Comment by NatalieEMoore** [comment]
Thanks @U-DON.  After launch we may want @amybehrens/ @angel to confirm that that is the best banner language!

File: src/js/messaging/utils/helpers.js (hunk 2 above)
Comment (U-DON) [inline src/js/messaging/utils/helpers.js]: Doesn't seem to work unless I pass `Promise` to `bind`. I guess some internal logic of `Promise.reject` relies on `this` being `Promise`.

**Pull request merged**